packages/aws-cdk/lib/api/aws-auth/sdk.ts

```diff
@@ -188,14 +189,26 @@ export class SDK implements ISDK {
   }: S3ClientOptions = {}): AWS.S3 {
     const config = { ...this.config };
 
-    if (!apiRequiresMd5Checksum) {
+    if (crypto.getFips() && apiRequiresMd5Checksum) {
+      // This should disappear for SDKv3; in SDKv3, we can always force the client to use SHA256 checksums
+      throw new Error('This operation requires MD5 for integrity purposes; unfortunately, it therefore not available in a FIPS environment.');
+    }
+
+    if (crypto.getFips()) {
```

Comment:
💅🏻 This might be slightly easier to undertand.

```suggestion
    if (crypto.getFips()) {
      if (apiRequiresMd5Checksum) {
        // This should disappear for SDKv3; in SDKv3, we can always force the client to use SHA256 checksums
        throw new Error('This operation requires MD5 for integrity purposes; unfortunately, it therefore not available in a FIPS environment.');
      }
```